tree.go

```diff
@@ -1533,9 +1563,12 @@ func (n *LeafNode) GetProofItems(keys keylist, _ NodeResolverFn) (*ProofElements
 		pe.Yis = append(pe.Yis, &leaves[0], &leaves[1])
 		pe.Fis = append(pe.Fis, suffPoly[:], suffPoly[:])
 		pe.Vals = append(pe.Vals, n.values[key[31]])
-		if len(esses) == 0 || esses[len(esses)-1] != extStatusPresent|(n.depth<<3) {
+
+		if _, ok := addedStems[string(key[:StemSize])]; !ok {
 			esses = append(esses, extStatusPresent|(n.depth<<3))
+			addedStems[string(key[:StemSize])] = struct{}{}
 		}
```

Comment:
This previous (deleted) `if` was wrong. Now we always need to add the extension status for the present stem. (That `len(esses) == 0` is not correct, since we could have proof of absence extensions status now).

In short, we simply add once the extension status present.

Reply:
ok yeah that makes sense